fip-4.md

```diff
@@ -0,0 +1,94 @@
+---
+fip: 4
+title: Provide ability to remove pub address from the FIO protocol for a user.
+status:Draft
+type: Functionality
+author: Ed Rotthoff <ed@dapix.io>
+created: 2020-04-16
+updated: 
+---
+
+## Abstract
+This FIP implements the following:
+* Adds new API end point and contract action for remove pub address.
+
+
+## Motivation
+Presently the FIO API does not provide any way for a user to remove public address mappings for a user:
+* It is foreseeable that users will have a need to remove pub address mappings
+* When a certain token is no longer supported
+* Whenever a user might desire to remove a certain mapping from state for privacy reasons.
+
+
+## Specification
+### Remove FIO address
+#### New end point: *remove_fio_address*
+##### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_address|Yes|fio address, see FIO address validation rules.|FIO Address which will have public addresses removed.|
+|public_addresses|Yes|JSON Array. See public_addresses below. Min: 1 item Max: 5 item|The public address to be added to the FIO Address for the specified token.|
```

Comment:
Why limit to 5? It only reduces state. Should we allow to cancel all?

Reply:
did this so the calls remain consistent and thus familiar....

Reply:
I think we need to either allow unlimited or have an ALL option, or perhaps both. This reduces state and should be encouraged.